File: Тестовые задания на циклы, условия, массивы, простые алгоритмы/3.py
def toTernaryString (n):
  if n == 0:
    return '0'
  nums = []
  while n:
    n, r = divmod(n, 3)
    nums.append(str(r))
  return ''.join(reversed(nums))

def toDecimal(num):
  result = 0
  lenth = len(num) - 1
  for i in range(len(num)):
    result += int(num[i]) * pow(3, lenth)
    lenth -= 1
  return result

num = int(input('Enter num: '))
ternaryStr = toTernaryString(num)

ONEMORESTRING = ''
for i in range (len(ternaryStr)):
  ONEMORESTRING += ternaryStr[i]
  ONEMORESTRING += ternaryStr[i]

result = toDecimal(ONEMORESTRING)

print(result)

File: Тестовые задания на циклы, условия, массивы, простые алгоритмы/3.cpp
#include <iostream>
#include <cmath>

using namespace std;

string toTernaryString(int num){
  string result;
  do{
    result = (to_string(num % 3)) + result;
    num = num / 3;
  }
  while (num > 0);

  return result;
}

int toDecimal(string num){
  int result = 0;
  int len = num.length() - 1;
  for (int i = 0; i < num.length(); i++){
    result += (int(num[i]) - 48) * pow(3, len);
    len--;
  }
  return result;
}

int main() {
  int num;
  
  cout << "enter number: ";
  cin >> num;

  string ternaryStr = toTernaryString(num);

  string ONEMORESTRING;
  for(int i = 0; i < ternaryStr.length(); i++){
    ONEMORESTRING += ternaryStr[i];
    ONEMORESTRING += ternaryStr[i];
  }

  int result = toDecimal(ONEMORESTRING);

  cout << result;
}